### ISSUE TITLE
input redirection from shell can cause an error on startup

### ISSUE THREAD

**Issue by bobisageek** [babashka/babashka#1700]
This is not a high priority to me or anything; just opening this since I stumbled across it.

**version**
1.3.190 (and also master)

**platform**

linux

**problem**

`bb <([some command here])` causes a NoSuchFileException (on my system, at least).

**repro**

```
$ bb <(echo '(println "hi")')
Exception in thread "main" java.nio.file.NoSuchFileException: /dev/fd/63
	at java.base@22/sun.nio.fs.UnixPath.toRealPath(UnixPath.java:886)
	at babashka.fs$real_path.invokeStatic(fs.cljc:80)
	at babashka.fs$real_path.invoke(fs.cljc:76)
	at babashka.fs$real_path.invokeStatic(fs.cljc:78)
	at babashka.main$resolve_symbolic_link.invokeStatic(main.clj:1144)
	at babashka.main$resolve_symbolic_link.invoke(main.clj:1143)
	at babashka.main$main.invokeStatic(main.clj:1171)
	at babashka.main$main.doInvoke(main.clj:1148)
...
```
**expected behavior**

The output of the shell command would be handled as a file that's 'ephemeral', e.g.
```
$ clj -M <(echo '(println "hi")')
hi
```

**notes**

After a bit of digging, this appears to stem from `babashka.main/main`:
https://github.com/babashka/babashka/blob/306842857803f272f0ac900f54710c61e5717b06/src/babashka/main.clj#L1181-L1182

... the redirection in use `<()` creates an ephemeral file (or something along those lines) which I guess doesn't exist by the time bb tries to get the real path. I'm not sure if there's a lot to really do about this without messing with #1661 , but I figure it couldn't hurt to open the discussion. This can generally be worked around by redirecting in a different way.

**Comment by bobisageek** [comment]
I confused my wording a bit... the file exists according to `fs/exists?`, but `toRealPath` throws the NoSuchFileException. This seems like it might be something at the JVM level, as I'm able to re-create the exception in bare-bones clj:
```clojure repl
$ clj -M -r --init <(echo '(println "hi")')
+ clj -M -r --init /dev/fd/63
++ echo '(println "hi")'
Clojure 1.11.3
user=> (.exists (.toFile (java.nio.file.Path/of "/dev/fd/63" (make-array String 0))))
true
user=> (.toRealPath (java.nio.file.Path/of "/dev/fd/63" (make-array String 0)) (make-array java.nio.file.LinkOption 0))
Execution error (NoSuchFileException) at sun.nio.fs.UnixException/translateToIOException (UnixException.java:92).
/dev/fd/63
```

`fs/canoncialize` does seem to work in my particular scenario, but there might be other implications there.

**Comment by borkdude** [comment]
canonicalize doesn't resolve symlinks AFAIK (I could be wrong) but just a try/catch around this could be worth it?

**Comment by bobisageek** [comment]
Canonicalizing following symlinks is maybe a system-dependent thing. In my environment it does, but on mac and Windows filesystem impls, idk. But `fs/canonicalize` takes an option for not following symlinks, which leads me to think it's sort of expected to follow them. Maybe I'll break out the windows machine over the weekend and experiment a little bit. I'm also gonna do some looking around and see if I can figure out why the file 'exists' but `real-path` doesn't like it.

**Comment by borkdude** [comment]
Fixed by #1702
